components/eam/src/physics/cam/gw_common.F90

```diff
@@ -742,4 +746,1472 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+
+subroutine dxygrid(dx,dy,theta_in,dxy)
+
+  IMPLICIT NONE
+  real(r8),intent(in) :: dx,dy,theta_in
+  real(r8),intent(out):: dxy
+  !local variables
+  real(r8) :: rad,theta,theta1
+
+  rad=4.0_r8*atan(1.0_r8)/180.0_r8
+  theta1=MOD(theta_in,360._r8)
+  !set negative axis into 0~360
+  if (theta1.ge.-360._r8.and.theta1.lt.0._r8) then
+  theta1=theta1+360._r8
+  endif
+  !in case the angle is not into the judgement
+  theta=theta1
+  !transform of angle into first quadrant
+  if      (theta1.ge.  0._r8.and.theta1.lt. 90._r8) then
+  theta=theta1
+  else if (theta1.gt. 90._r8.and.theta1.lt.180._r8) then
+  theta=(180._r8-theta1)
+  else if (theta1.gt.180._r8.and.theta1.lt.270._r8) then
+  theta=(theta1-180._r8)
+  else if (theta1.gt.270._r8.and.theta1.lt.360._r8) then
+  theta=(360._r8-theta1)
+  else if (theta1.eq.90._r8.or.theta1.eq.270._r8) then
+  theta=90._r8
+  else if (theta1.eq.0._r8.or.theta1.eq.180._r8) then
+  theta=0._r8
+  endif
+
+  !get dxy
+  if   (theta.ge. 0._r8.and.theta.lt.atan2(dy,dx)/rad) then
+  dxy=dx/cos(theta*rad)
+  else if (theta.ge.atan2(dy,dx)/rad.and.theta.le.90._r8)then
+  dxy=dy/sin(theta*rad)
+  endif
+
+end subroutine dxygrid
+
+!==========================================================================
+
+subroutine grid_size(state, grid_dx, grid_dy)
+  ! Determine the size of the grid for each of the columns in state
+
+  use phys_grid,       only: get_area_p
+  use shr_const_mod,   only: shr_const_pi
+  use physics_types,   only: physics_state
+  use ppgrid,          only: pver, pverp, pcols
+
+  type(physics_state), intent(in) :: state
+  real(r8), intent(out)           :: grid_dx(pcols), grid_dy(pcols)   ! E3SM grid [m]
+
+  real(r8), parameter :: earth_ellipsoid1 = 111132.92_r8 ! World Geodetic System 1984 (WGS84) 
+                                                         ! first coefficient, meters per degree longitude at equator
+  real(r8), parameter :: earth_ellipsoid2 = 559.82_r8 ! second expansion coefficient for WGS84 ellipsoid
+  real(r8), parameter :: earth_ellipsoid3 = 1.175_r8 ! third expansion coefficient for WGS84 ellipsoid
+  real(r8) :: mpdeglat, column_area, degree, lat_in_rad
+  integer  :: i
+  
+  do i=1,state%ncol
+      ! determine the column area in radians
+      column_area = get_area_p(state%lchnk,i)
+      ! convert to degrees
+      degree = sqrt(column_area)*(180._r8/shr_const_pi)
+
+      ! convert latitude to radians
+      lat_in_rad = state%lat(i)*(shr_const_pi/180._r8)
+
+      ! Now find meters per degree latitude
+      ! Below equation finds distance between two points on an ellipsoid, derived from expansion
+      !  taking into account ellipsoid using World Geodetic System (WGS84) reference 
+      mpdeglat = earth_ellipsoid1 - earth_ellipsoid2 * cos(2._r8*lat_in_rad) + earth_ellipsoid3 * cos(4._r8*lat_in_rad)
+      grid_dx(i) = mpdeglat * degree
+      grid_dy(i) = grid_dx(i) ! Assume these are the same
+  enddo
+end subroutine grid_size
```

Comment:
From back when I implemented the physgrid infrastructure I know we already calculate this elsewhere. Would it be possible to use that area calculation for consistency?

Reply:
Not sure. I know this is an estimate. But using other ways of calculation may change a little results. Depends on how different your calculation is from this calculation.